Automattic/studio
# Studio: Add chat rating feedback feature

- Closes https://github.com/Automattic/dotcom-forge/issues/8980

## Proposed Changes

This PR adds a rating system to the chat assistant so that the user can leave feedback. Initial view:

<img width="807" alt="Screenshot 2024-09-12 at 11 26 24 AM" src="https://github.com/user-attachments/assets/958f1234-927c-4841-b028-1d990713fb53">

After the feedback has been left:

<img width="1163" alt="Screenshot 2024-09-12 at 11 26 53 AM" src="https://github.com/user-attachments/assets/b147842f-b445-4cb8-bf2c-1bc107fa0495">

`Thanks for the feedback` notice should persist once you click it on a specific message.

## Testing Instructions

* Apply D161215-code if not yet deployed
* Pull the changes from this branch
* Start the app with `nvm use && npm install && npm start`
* Navigate to the Assistant tab
* Send a message and wait for assistant response
* Once the message is sent, observe the rating to appear
* Open the `Network` tab in the browser console
* Click on either positive or negative rating
* Observe `Thanks for the feedback` notice
* Onserve in the Network tab that API request succeeds (look for `feedback`):

<img width="1421" alt="Screenshot 2024-09-12 at 11 31 10 AM" src="https://github.com/user-attachments/assets/7baefb68-7ca8-4fc5-9d60-8235c539bdb4">

* Note down the `chatId` parameter:

<img width="1227" alt="Screenshot 2024-09-12 at 11 32 24 AM" src="https://github.com/user-attachments/assets/ad23c894-217f-4f7e-a3ca-b638299e47ac">

* Open the developer console on: https://developer.wordpress.com/docs/api/console/
* Query the following endpoint `wpcom/v2/`, `GET`, `/odie/chat/wpcom-studio-chat/${chatId}` (use chatId that you noted down earlier)

<img width="1335" alt="Screenshot 2024-09-12 at 11 34 30 AM" src="https://github.com/user-attachments/assets/6cb14478-7c55-436e-a0b5-288485e72895">

* Look for the message that you rated and confirm the rating is present (you can find it by the messageId from the API request):

<img width="1325" alt="Screenshot 2024-09-12 at 11 36 04 AM" src="https://github.com/user-attachments/assets/d3ec9288-629e-47dd-9458-47f4e9a42385">

* Try sending a couple of other messages and rating them
* Confirm that `Thanks for the feedback` notice persists

## Pre-merge Checklist
